src/Filesystem/TemporaryDirectoryManager.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Ingenerator\PHPUtils\Filesystem;
+
+
+use InvalidArgumentException;
+use function array_keys;
+use function escapeshellarg;
+
+class TemporaryDirectoryManager
+{
+    /**
+     * @var boolean[]
+     */
+    protected array $directories = [];
+    public function __construct(protected string $base_dir)
+    {
+    }
+
+    /**
+     * Create a temporary directory with a unique name under this prefix. The directory
+     * will be deleted when the manager is destroyed, if it still exists. Does not include trailing slash!
+     */
+    public function mkTemp(string $prefix): string
+    {
+        $path = $this->base_dir.'/'.\uniqid($prefix, TRUE);
+
+        try {
+            if ( ! mkdir($path, 0700, TRUE) && ! is_dir($path)) {
+                throw new \ErrorException('Could not make '.$path.': reason unknown');
+            }
+        } catch (\ErrorException $e) {
+            throw new \RuntimeException('Could not make temporary directory: '.$e->getMessage());
+        }
+
+        $this->directories[$path] = TRUE;
+
+        return $path;
+    }
+
+    /**
+     * Explicitly clean up the managed directory
+     */
+    public function cleanup(string $path): void
+    {
+        if ( ! isset($this->directories[$path])) {
+            throw new InvalidArgumentException('Cannot cleanup `'.$path.'` - not a managed directory');
+        }
+
+        $this->removeDirectory($path);
+    }
+
+    protected function removeDirectory(string $path): void
+    {
+        \exec('rm -rf '.escapeshellarg($path));
```

Comment:
Do we want to be any stricter about this implementation e.g. throwing if it fails or etc?

Reply:
I don't know. It seems a bit clunky just to recursively delete a directory but there isn't anything cleaner that's dependency free. Do we want it to throw inside of a destructor anyway? It works for the purposes we're using it for, I'm tempted to say improving on it is for another day? WDYT

Reply:
Fair point, yes I agree I'd favour destructing clean and dealing with any failures (which basically shouldn't happen anyway unless someone's done something super weird like mess with the permissions while the dir was in use) separately/never